File: src/main/java/org/pixeltime/enchantmentsenhance/event/enchantment/Auto_Block.java
package org.pixeltime.enchantmentsenhance.event.enchantment;

import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockBreakEvent;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class Auto_Block implements Listener {
    public static void removeItems(final Inventory inventory, final Material material, int n) {
        if (n <= 0) {
            return;
        }
        for (int size = inventory.getSize(), i = 0; i < size; ++i) {
            final ItemStack item = inventory.getItem(i);
            if (item != null) {
                if (material == item.getType()) {
                    final int amount = item.getAmount() - n;
                    if (amount > 0) {
                        item.setAmount(amount);
                        break;
                    }
                    inventory.clear(i);
                    n = -amount;
                    if (n == 0) {
                        break;
                    }
                }
            }
        }
    }

    @EventHandler
    public void onBreak(final BlockBreakEvent blockBreakEvent) {
        final Player player = blockBreakEvent.getPlayer();
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "auto_block"));
        if (player.getInventory().getItemInHand() != null && player.getItemInHand().hasItemMeta() && player.getItemInHand().getItemMeta().hasLore() && player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
            this.autoBlock(player);
        }
    }

    public void autoBlock(final Player player) {
        try {
            ItemStack[] contents;
            for (int length = (contents = player.getInventory().getContents()).length, i = 0; i < length; ++i) {
                final ItemStack itemStack = contents[i];
                final int n = itemStack.getAmount() / 9;
                final int n2 = itemStack.getAmount() / 4;
                if (n > 0) {
                    if (itemStack.getType() == Material.EMERALD) {
                        removeItems(player.getInventory(), Material.EMERALD, n * 9);
                        player.getInventory().addItem(new ItemStack[]{new ItemStack(Material.EMERALD_BLOCK, n)});
                    }
                    if (itemStack.getType() == Material.DIAMOND) {
                        removeItems(player.getInventory(), Material.DIAMOND, n * 9);
                        player.getInventory().addItem(new ItemStack[]{new ItemStack(Material.DIAMOND_BLOCK, n)});
                    }
                    if (itemStack.getType() == Material.IRON_INGOT) {
                        removeItems(player.getInventory(), Material.IRON_INGOT, n * 9);
                        player.getInventory().addItem(new ItemStack[]{new ItemStack(Material.IRON_BLOCK, n)});
                    }
                    if (itemStack.getType() == Material.GOLD_INGOT) {
                        removeItems(player.getInventory(), Material.GOLD_INGOT, n * 9);
                        player.getInventory().addItem(new ItemStack[]{new ItemStack(Material.GOLD_BLOCK, n)});
                    }
                    if (itemStack.getType() == Material.INK_SACK && itemStack.getData().getData() == 4) {
                        removeItems(player.getInventory(), Material.INK_SACK, n * 9);
                        player.getInventory().addItem(new ItemStack[]{new ItemStack(Material.LAPIS_BLOCK, n)});
                    }
                    if (itemStack.getType() == Material.COAL) {
                        removeItems(player.getInventory(), Material.COAL, n * 9);
                        player.getInventory().addItem(new ItemStack[]{new ItemStack(Material.COAL_BLOCK, n)});
                    }
                    if (itemStack.getType() == Material.REDSTONE) {
                        removeItems(player.getInventory(), Material.REDSTONE, n * 9);
                        player.getInventory().addItem(new ItemStack[]{new ItemStack(Material.REDSTONE_BLOCK, n)});
                    }
                }
                if (n2 > 0) {
                    if (itemStack.getType() == Material.CLAY_BALL) {
                        removeItems(player.getInventory(), Material.CLAY_BALL, n * 4);
                        player.getInventory().addItem(new ItemStack[]{new ItemStack(Material.CLAY, n)});
                    }
                    if (itemStack.getType() == Material.SNOW_BALL) {
                        removeItems(player.getInventory(), Material.SNOW_BALL, n * 9);
                        player.getInventory().addItem(new ItemStack[]{new ItemStack(Material.SNOW, n)});
                    }
                }
            }
        } catch (Exception ex) {
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantment/Auto_Block.kt
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.pixeltime.enchantmentsenhance.event.enchantment

import org.bukkit.ChatColor
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockBreakEvent
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack
import org.pixeltime.enchantmentsenhance.manager.IM
import org.pixeltime.enchantmentsenhance.manager.KM
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Auto_Block : Listener {
    private val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "auto_block"))

    @EventHandler
    fun onBreak(blockBreakEvent: BlockBreakEvent) {
        val player = blockBreakEvent.player
        val armorContents = player.inventory.armorContents + IM.getAccessorySlots(player)
        for (i in armorContents.indices) {
            val itemStack = armorContents[i]
            if (itemStack.hasItemMeta() && itemStack.itemMeta.hasLore()) {
                val level = KM.getLevel(translateAlternateColorCodes, itemStack.itemMeta.lore)
                if (level > 0) {
                    this.autoBlock(player)
                }
            }
        }
    }

    fun autoBlock(player: Player) {
        try {
            val contents: Array<ItemStack> = player.inventory.contents
            for (i in contents.indices) {
                val itemStack = contents[i]
                val n = itemStack.amount / 9
                val n2 = itemStack.amount / 4
                if (n > 0) {
                    if (itemStack.type == Material.EMERALD) {
                        removeItems(player.inventory, Material.EMERALD, n * 9)
                        player.inventory.addItem(ItemStack(Material.EMERALD_BLOCK, n))
                    }
                    if (itemStack.type == Material.DIAMOND) {
                        removeItems(player.inventory, Material.DIAMOND, n * 9)
                        player.inventory.addItem(ItemStack(Material.DIAMOND_BLOCK, n))
                    }
                    if (itemStack.type == Material.IRON_INGOT) {
                        removeItems(player.inventory, Material.IRON_INGOT, n * 9)
                        player.inventory.addItem(ItemStack(Material.IRON_BLOCK, n))
                    }
                    if (itemStack.type == Material.GOLD_INGOT) {
                        removeItems(player.inventory, Material.GOLD_INGOT, n * 9)
                        player.inventory.addItem(ItemStack(Material.GOLD_BLOCK, n))
                    }
                    if (itemStack.type == Material.INK_SACK && itemStack.data.data.toInt() == 4) {
                        removeItems(player.inventory, Material.INK_SACK, n * 9)
                        player.inventory.addItem(ItemStack(Material.LAPIS_BLOCK, n))
                    }
                    if (itemStack.type == Material.COAL) {
                        removeItems(player.inventory, Material.COAL, n * 9)
                        player.inventory.addItem(ItemStack(Material.COAL_BLOCK, n))
                    }
                    if (itemStack.type == Material.REDSTONE) {
                        removeItems(player.inventory, Material.REDSTONE, n * 9)
                        player.inventory.addItem(ItemStack(Material.REDSTONE_BLOCK, n))
                    }
                }
                if (n2 > 0) {
                    if (itemStack.type == Material.CLAY_BALL) {
                        removeItems(player.inventory, Material.CLAY_BALL, n * 4)
                        player.inventory.addItem(ItemStack(Material.CLAY, n))
                    }
                    if (itemStack.type == Material.SNOW_BALL) {
                        removeItems(player.inventory, Material.SNOW_BALL, n * 9)
                        player.inventory.addItem(ItemStack(Material.SNOW, n))
                    }
                }
            }
        } catch (ex: Exception) {
        }

    }

    companion object {
        fun removeItems(inventory: Inventory, material: Material, num: Int) {
            var n = num
            if (n <= 0) {
                return
            }
            val size = inventory.size
            var i = 0
            while (i < size) {
                val item = inventory.getItem(i)
                if (item != null) {
                    if (material == item.type) {
                        val amount = item.amount - n
                        if (amount > 0) {
                            item.amount = amount
                            break
                        }
                        inventory.clear(i)
                        n = -amount
                        if (n == 0) {
                            break
                        }
                    }
                }
                ++i
            }
        }
    }
}
